walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -64,6 +64,29 @@ else if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
 
 
         for (Event event2 : events){
+
+            //optional attendee in every attendee considered 
+            for (Event event3 : events){//new
+                if (event1 != event2 && event2!=event3){//new
+                    if (!event1.getWhen().overlaps(event2.getWhen())){
```

Comment:
These parts here don't deal with `event3`, maybe think about if you can move them out of the event3 loop.  

Or maybe get rid of the event3 loop all together to improve efficiency.  Imagine if you have 1000 events.  Then 3 nested loops would evaluate this section here 1000x1000x1000 times.


Definitely ok to start with the 3 loops then refactor to optimize later.  Although it could be helpful to think about what the differences are between the 3 loops, and whether or not they are related to each other.  ie if loop3 deals with optional attendees, maybe you can finish up with loops 1 and loop 2 for mandatory attendees, then do a separate loop for optional attendees?